### PR TITLE
#17: added methods for puting users in database

### PR DESCRIPTION
**Pull request by Mattihew**

### CODE REVIEW

File: YHP_website/src/main/java/storage/UserCache.java [hunk 1]
```diff
@@ -36,46 +38,67 @@ private UserCache(final Database database)
...
 			e.printStackTrace();
 		}
-		if (users == null || users.isEmpty())
+		if (foundUsers == null || foundUsers.isEmpty())
 		{
 			return null;
 		}
-		return users.get(0);
+		this.users.addAll(foundUsers);
+		return foundUsers.get(0);
```
Comment (jsjlewis96) [inline YHP_website/src/main/java/storage/UserCache.java]: Not too keen on this only returning one user, unless I'm misinterpreting the purpose of this method I think this should return all users where the query is true

File: YHP_website/src/main/java/storage/UserCache.java [hunk 2]
```diff
@@ -36,46 +38,67 @@ private UserCache(final Database database)
...
+	}
+	
+	public void putUser(final User user) throws SQLException
+	{
+		AsyncQueryRunner run = this.database.getAsyncQueryRunner();
+		run.update("INSERT INTO users (user_id, user_name, user_pass, forename, surname, address_id) VALUES (?,?,?,?,?,?)"
```
Comment (jsjlewis96) [inline YHP_website/src/main/java/storage/UserCache.java]: Could this use UPSERT? Also a comment explainging the possible results would be nice

File: YHP_website/src/main/java/storage/UserCache.java [hunk 3]
```diff
@@ -36,46 +38,67 @@ private UserCache(final Database database)
...
+			userBuilder.digest(new Digest(rs.getString("user_pass")));
+			userBuilder.id(UUID.fromString(rs.getString("user_id")));
 			final List<String> roles = roleRunner.query(
-					"SELECT role_name FROM user_roles WHERE user_name='?';",
+					"SELECT role_name FROM user_roles WHERE user_name=?;",
 					new ColumnListHandler<String>(), rs.getString("user_name"));
-			userBuilder.id(UUID.fromString(rs.getString("user_id")));
 			userBuilder.role(UserRole.fromDatabaseValues(roles));
 			// userBuilder.address(value); TODO add address getter.
 			return userBuilder.build();
```
Comment (jsjlewis96) [inline YHP_website/src/main/java/storage/UserCache.java]: I can't immediately tell what this all does, a little explanation would be handy

File: YHP_website/src/main/java/storage/UserCache.java (hunk 1 above)
Comment (Mattihew) [inline YHP_website/src/main/java/storage/UserCache.java]: agree. changed this method to return all.

File: YHP_website/src/main/java/storage/UserCache.java (hunk 2 above)
Comment (Mattihew) [inline YHP_website/src/main/java/storage/UserCache.java]: The `ON CONFLICT ... DO UPDATE` is the UPSERT.
added javadoc explanation

File: YHP_website/src/main/java/storage/UserCache.java (hunk 3 above)
Comment (Mattihew) [inline YHP_website/src/main/java/storage/UserCache.java]: added some comments. let me know if more is needed.

**Pull request merged**